webapp/src/components/sidebar_right/sidebar_right.tsx

```diff
@@ -0,0 +1,208 @@
+            return true;
+        }
+    }
+}
+
+export default class SidebarRight extends React.PureComponent<PropTypes> {
+
+  componentDidMount() {
+      if (this.props.yourPrs && this.props.rhsState === RHSStates.PRS) {
+          this.props.actions.getYourPrDetails(
+              mapGitlabItemListToPrList(this.props.yourPrs),
+          );
+      }
+      if (this.props.reviews && this.props.rhsState === RHSStates.REVIEWS) {
+          this.props.actions.getReviewDetails(
+              mapGitlabItemListToPrList(this.props.reviews),
+          );
+      }
+  }
+
+  componentDidUpdate(prevProps: PropTypes) {
+      if (
+          shouldUpdateDetails(
+              this.props.yourPrs,
+              prevProps.yourPrs,
+              RHSStates.PRS,
+              this.props.rhsState,
+              prevProps.rhsState,
+          )
+      ) {
+          this.props.actions.getYourPrDetails(
+              mapGitlabItemListToPrList(this.props.yourPrs),
+          );
+      }
+
+      if (
+          shouldUpdateDetails(
+              this.props.reviews,
+              prevProps.reviews,
+              RHSStates.REVIEWS,
+              this.props.rhsState,
+              prevProps.rhsState,
+          )
+      ) {
+          this.props.actions.getReviewDetails(
+              mapGitlabItemListToPrList(this.props.reviews),
+          );
+      }
+  }
+
+  render() {
+      const style = getStyle(this.props.theme)
+      const baseURL:string = this.props.gitlabURL ?? 'https://gitlab.com';
+      const orgQuery:string = this.props.org ? `+org%3A ${this.props.org}` : '';
+    
+      let title:string = '';
+      let gitlabItems: Item[] = [];
+      let listUrl:string = '';
+
+      switch (this.props.rhsState) {
+      case RHSStates.PRS:
+          gitlabItems = this.props.yourPrs;
+          title = 'Your Open Merge Requests';
+          listUrl = `${baseURL}/dashboard/merge_requests?state=opened&scope=all&author_username=${this.props.username}&archived=false${orgQuery}`;
+          break;
+      case RHSStates.REVIEWS:
+          gitlabItems = this.props.reviews;
+          listUrl = `${baseURL}/dashboard/merge_requests?state=opened&scope=all&assignee_username=${this.props.username}&archived=false${orgQuery}`;
+          title = 'Merge Requests Needing Review';
+          break;
+      case RHSStates.UNREADS:
+          gitlabItems = this.props.unreads;
+          title = 'Unread Messages';
+          listUrl = `${baseURL}/dashboard/todos`;
+          break;
+      case RHSStates.ASSIGNMENTS:
+          gitlabItems = this.props.yourAssignments;
+          title = 'Your Assignments';
+          listUrl = `${baseURL}/dashboard/issues?state=opened&scope=all&assignee_username=${this.props.username}${orgQuery}`;
+          break;
+      default:
+          break;
+      }
+
+      return (
+          <React.Fragment>
+              <Scrollbars
+                  autoHide={true}
+                  autoHideTimeout={500}     // Hide delay in ms
```

Comment:
it is always good practice to put this values in constants with meaningful names. Normally code changes but comments remain the same and that leads to confusion.